post-policy.go

```diff
@@ -302,11 +302,31 @@ func (p *PostPolicy) SetUserMetadata(key, value string) error {
 }
 
 // SetChecksum sets the checksum of the request.
-func (p *PostPolicy) SetChecksum(c Checksum) {
+func (p *PostPolicy) SetChecksum(c Checksum) error {
 	if c.IsSet() {
 		p.formData[amzChecksumAlgo] = c.Type.String()
 		p.formData[c.Type.Key()] = c.Encoded()
+
+		// Needed for S3 compatibility. MinIO ignores the checksum keys in the policy.
+		// https://github.com/minio/minio/blob/RELEASE.2024-08-29T01-40-52Z/cmd/postpolicyform.go#L60-L65
```

Comment:
As the comment mentions, this is needed for S3 compat, but I really am not sure why we ignore these keys in the MinIO implementation.

Reply:
Probably just a missing implementation, or it may have added later. Doesn't really matter. Should be trivial to add.

Reply:
(also , I don't know if AWS added that, but adding trailing checksums in forms should also be doable)